keyvi/include/keyvi/vector/vector_file.h

```diff
@@ -0,0 +1,147 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  vector_file.h
+ *
+ *  Created on: March 17, 2018
+ *      Author: Narek Gharibyan <narekgharibyan@gmail.com>
+ */
+
+#ifndef KEYVI_VECTOR_VECTOR_FILE_H_
+#define KEYVI_VECTOR_VECTOR_FILE_H_
+
+#include <exception>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include "dictionary/fsa/internal/value_store_factory.h"
+#include "util/serialization_utils.h"
+#include "vector/types.h"
+
+static const char KEYVI_VECTOR_BEGIN[] = "KEYVI_VECTOR_BEGIN";
+static const size_t KEYVI_VECTOR_BEGIN_LEN = 18;
+static const char KEYVI_VECTOR_END[] = "KEYVI_VECTOR_END";
+static const size_t KEYVI_VECTOR_END_LEN = 16;
+static const char MANIFEST[] = "manifest";
+static const char SIZE[] = "size";
+static const char VALUE_STORE_TYPE[] = "value_store_type";
+
+namespace keyvi {
+namespace vector {
+
+template <value_store_t>
+class Vector;
+
+class VectorFile {
+  template <value_store_t>
+  friend class Vector;
+
+  using mapped_region = boost::interprocess::mapped_region;
+  using IValueStoreReader = dictionary::fsa::internal::IValueStoreReader;
+
+ public:
+  explicit VectorFile(const std::string& filename, const value_store_t value_store_type) {
+    const auto loading_strategy = dictionary::loading_strategy_types::lazy_no_readahead;
+    std::ifstream in_stream(filename, std::ios::binary);
+
+    CheckValidity(&in_stream);
+
+    in_stream.seekg(KEYVI_VECTOR_BEGIN_LEN);
+    const auto file_ptree = util::SerializationUtils::ReadJsonRecord(in_stream);
+    if (value_store_type != boost::lexical_cast<size_t>(file_ptree.get<std::string>(VALUE_STORE_TYPE))) {
+      throw std::invalid_argument("wrong vector file");
+    }
+    manifest_ = file_ptree.get<std::string>(MANIFEST);
+
+    const auto index_ptree = util::SerializationUtils::ReadJsonRecord(in_stream);
+    size_ = boost::lexical_cast<size_t>(index_ptree.get<std::string>(SIZE));
+    const auto index_size = size_ * sizeof(offset_type);
+
+    auto file_mapping = boost::interprocess::file_mapping(filename.c_str(), boost::interprocess::read_only);
+    const auto map_options = dictionary::fsa::internal::MemoryMapFlags::FSAGetMemoryMapOptions(loading_strategy);
+    index_region_ = mapped_region(file_mapping, boost::interprocess::read_only, in_stream.tellg(), index_size, nullptr,
+                                  map_options);
+    index_region_.advise(mapped_region::advice_types::advice_random);
```

Comment:
this is not make use of loading strategy, you could use FSAGetMemoryMapAdvices(loading_strategy)